doc/src/bond_cpp.md

```diff
@@ -1271,6 +1271,37 @@ See examples:
 - `examples/cpp/core/simple_json`
 
 
+Protocol Buffers Interop
+========================
+
+Bond provides a `ProtobufBinaryWriter` that can be used as any other built-in
+protocol writer in order to serialize objects to Protocol Buffers binary format.
+It requires a compatible .bond schema definition that will map to corresponding
+.proto schema:
+
+- Field ordinals must match.
+- The `bool`, `enum`, `uint32`, `uint64`, `int32`, `int64`, `float`,
+  `double`, and `string` types are mapped to corresponding ones with same name.
+- The `bytes` is represented as a `blob`.
+- The `fixed32`, `fixed64`, `sfixed32` and `sfixed64` types can be expressed
+  as `uint32`, `uint64`, `int32` and `int64` respectively, but require an
+  additional `ProtoEncode("Fixed")` attribute.
+- The `sint32` and `sint64` types correspond to `int32` and `int64` with
+  additional `ProtoEncode("ZigZag")` attribute.
+- The `repeated` fields can be expressed as a `vector`, `list` or `set` which
+  will be packed for scalar numeric types. For unpacked encoding, an additional
+  `ProtoPack("False")` attribute must be used.
+- The `map` requires different `ProtoEncodeKey` and `ProtoEncodeValue`
+  attributes for numeric key/value types with `Fixed` and `ZigZag` encoding.
+- Since Bond uses a list representation on the wire for `nullable<T>`, it can
+  also be used to represent a scalar field (`optional` for `proto2`) or a
+  `repeated` one which is expected to have at most one element. Note, that in
+  case of a numeric scalar field or an unpacked `repeated` numeric type, the
+  `ProtoPack("False")` must be used.
+- The smaller 8 and 16 bit numeric Bond types can also be mapped to 32 or 64 bit
+  ones with corresponding encoding.
+
+
```

Comment:
We could link to an example/test here like in the rest of documentation.

Reply:
Will add an example as well. Looks like both examples and tests are tied to `BOND_SKIP_CORE_TESTS` (was trying to avoid unnecessarily build protobuf).